frontend/src/app/App.tsx

```diff
@@ -42,80 +43,79 @@ const App: React.FC = () => {
     loadError: fetchConfigError,
   } = useApplicationSettings();
 
+  const [storageClasses] = useStorageClasses();
+
   useDetectUser();
 
-  // We lack the critical data to startup the app
-  if (userError || fetchConfigError) {
-    // There was an error fetching critical data
+  if (!username || !configLoaded || !dashboardConfig) {
+    // We lack the critical data to startup the app
+    if (userError || fetchConfigError) {
+      // There was an error fetching critical data
+      return (
+        <Page>
+          <PageSection>
+            <Stack hasGutter>
+              <StackItem>
+                <Alert variant="danger" isInline title="General loading error">
+                  <p>
+                    {(userError ? userError.message : fetchConfigError?.message) ||
+                      'Unknown error occurred during startup.'}
+                  </p>
+                  <p>Logging out and logging back in may solve the issue.</p>
+                </Alert>
+              </StackItem>
+              <StackItem>
+                <Button
+                  variant="secondary"
+                  onClick={() => logout().then(() => window.location.reload())}
+                >
+                  Logout
+                </Button>
+              </StackItem>
+            </Stack>
+          </PageSection>
+        </Page>
+      );
+    }
+
+    // Assume we are still waiting on the API to finish
     return (
-      <Page>
-        <PageSection>
-          <Stack hasGutter>
-            <StackItem>
-              <Alert variant="danger" isInline title="General loading error">
-                <p>
-                  {(userError ? userError.message : fetchConfigError?.message) ||
-                    'Unknown error occurred during startup.'}
-                </p>
-                <p>Logging out and logging back in may solve the issue.</p>
-              </Alert>
-            </StackItem>
-            <StackItem>
-              <Button
-                variant="secondary"
-                onClick={() => logout().then(() => window.location.reload())}
-              >
-                Logout
-              </Button>
-            </StackItem>
-          </Stack>
-        </PageSection>
-      </Page>
+      <Bullseye>
+        <Spinner />
+      </Bullseye>
     );
   }
 
-  // Waiting on the API to finish
-  const loading = !username || !configLoaded || !dashboardConfig;
-
   return (
-    <AreaContextProvider>
-      {loading ? (
-        <Bullseye>
-          <Spinner />
-        </Bullseye>
-      ) : (
-        <AppContext.Provider
-          value={{
-            buildStatuses,
-            dashboardConfig,
-          }}
+    <AppContext.Provider
+      value={{
```

Comment:
You got a conflict resolution issue here -- @christianvogt moved AreaContext up for speed reasons.

Reply:
You might want to compare the branch and `main` during conflicts to see what is what -- `main` almost always will win unless it's directly related to the change this branch was after (storageClasses related).

You could tri-compare with incubation to try and get more, but that might not work out the best. But could give more context.

Reply:
interesting and nice to see this is finding its way into main / downstream.
I am currently lookin' through to get an overview.
Let me know in case any questions pup up. What I can say: this feature works well in incubation. The idea is to put the cluster's storageClasses array holding all related information per storageClass into appContext to retrieve later.

Reply:
@andrewballantyne why is Christian's change https://github.com/opendatahub-io/odh-dashboard/commit/d995c1c581095c379f1615c7e26fb46ebc517a4c in main, but not in incubation yet?

Reply:
Updated - thanks for the useful feedback.

Reply:
@shalberd it's possible the merge to incubation didn't go smoothly (assuming incubation is up date). It's been a cause of some friction process wise and something we're actively discussing how to improve right now.

Reply:
yes, the way it's now is just right, thank you for having updated that.

Reply:
Incubation has had rough goings as of late, we are looking to simplify everything branch-wise and I have not been super updating about it because the conflicts are a mess each time I go to do it.

It is most definitely not up to date for the last release or so -- it got mid-last release some Pipeline Versions & Logs through the work we were actively working on, and whatever main was at that time... but since then it's been cherry picks of critical fixes and that's about it.